hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/ExchangeRatePrecompiledContract.java

```diff
@@ -0,0 +1,99 @@
+package com.hedera.services.store.contracts.precompile;
+
+import com.esaulpaugh.headlong.abi.BigIntegerType;
+import com.esaulpaugh.headlong.abi.TypeFactory;
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.fees.HbarCentExchange;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.evm.Gas;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.precompile.AbstractPrecompiledContract;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.math.BigInteger;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+/**
+ * System contract to interconvert tinybars and tinycents at the active exchange rate.
+ * The ABI consists of 1 to 9 packed bytes where,
+ * <ol>
+ *    <li>The first byte is either {@code 0xbb}, when converting to tinybars; or
+ *    {@code 0xcc}, when converting to tinycents.</li>
+ *    <li>The remaining 0 to 8 bytes are (logically) left-padded with zeros to
+ *    form an eight-byte big-endian representation of a {@code long} value.</li>
+ * </ol>
+ *
+ * <p> When the input {@code Bytes} take this form, <i>and</i> the given value can
+ * be converted to the requested denomination without over-flowing an eight-byte
+ * value, the contract returns the conversion result. Otherwise, it returns null.
+ */
+@Singleton
+public class ExchangeRatePrecompiledContract extends AbstractPrecompiledContract {
+	private static final String PRECOMPILE_NAME = "ExchangeRate";
+	private static final BigIntegerType WORD_DECODER = TypeFactory.create("uint256");
+
+	static final Bytes INVALID_CALL_REVERT_REASON = Bytes.of("Invalid call".getBytes());
+
+	//tinycentsToTinybars(uint256)
+	static final int TO_TINYBARS_SELECTOR = 0x2e3cff6a;
+	//tinybarsToTinycents(uint256)
+	static final int TO_TINYCENTS_SELECTOR = 0x43a88229;
+
+	public static final String EXCHANGE_RATE_SYSTEM_CONTRACT_ADDRESS = "0x21d4";
```

Comment:
We can't use that address.  it could collied with `0.0.8660`. it needs to be between 500 and 999

Reply:
Ugh, too cute by half---updated to `0x168`.